ror-tools/src/test/scala/tech/beshu/ror/tools/RorToolsAppSuite.scala

```diff
@@ -0,0 +1,231 @@
+          |Patching ...
+          |Elasticsearch is patched! ReadonlyREST is ready to use"""
+          .stripMargin
+      )
+      val hashAfterPatching = DirectoryUtils.calculateHash(esLocalPath)
+
+      // Verify
+      val (verifyResult, verifyOutput) = captureResultAndOutput {
+        RorToolsAppHandler.handle(Array("verify", "--es-path", esLocalPath.toString))(_)
+      }
+
+      verifyResult should equal(Result.Success)
+      verifyOutput should include(
+        """Checking if Elasticsearch is patched ...
+          |Elasticsearch is patched! ReadonlyREST can be used"""
+          .stripMargin
+      )
+
+      // Unpatch
+      val hashBeforeUnpatching = DirectoryUtils.calculateHash(esLocalPath)
+      val (unpatchResult, unpatchOutput) = captureResultAndOutput {
+        RorToolsAppHandler.handle(Array("unpatch", "--es-path", esLocalPath.toString))(_)
+      }
+      unpatchResult should equal(Result.Success)
+      unpatchOutput should include(
+        """Checking if Elasticsearch is patched ...
+          |Restoring ...
+          |Elasticsearch is unpatched! ReadonlyREST can be removed now"""
+          .stripMargin
+      )
+      val hashAfterUnpatching = DirectoryUtils.calculateHash(esLocalPath)
+
+      hashBeforePatching should equal(hashAfterUnpatching)
+      hashAfterPatching should equal(hashBeforeUnpatching)
+      hashBeforePatching shouldNot equal(hashAfterPatching)
+    }
+  }
+
+  private def captureResultAndOutput(block: InOut => Result): (Result, String) = {
+    val inOut = new CapturingOutputAndMockingInput()
+    val result = block(inOut)
+    (result, inOut.getOutputBuffer)
+  }
+
+  private def captureResultAndOutputWithInteraction(block: InOut => Result, response: String): (Result, String) = {
+    val inOut = new CapturingOutputAndMockingInput(Some(response))
+    val result = block(inOut)
+    (result, inOut.getOutputBuffer)
+  }
+
+  private def withTestEsContainer(withStartedEs: EsContainer => Unit): Unit = {
+    val manager = new TestEsContainerManager
+    try {
+      manager.start()
+        .map(_ => withStartedEs(manager.esContainer))
+        .runSyncUnsafe(5 minutes)
+    } finally {
+      manager.stop().runSyncUnsafe()
+    }
+  }
+
+  private def unTar(tarPath: Path, outputPath: Path): Unit = {
+    val tarFile = new TarFile(tarPath.toFile)
+    for (entry <- tarFile.getEntries.asScala) {
+      val path = outputPath.resolve(entry.getName)
+      if (entry.isDirectory) {
+        Files.createDirectories(path)
+      } else {
+        Files.createDirectories(path.getParent)
+        Files.copy(tarFile.getInputStream(entry), path)
+      }
+    }
+  }
+
+  private def withFreshEsDirectory(perform: () => Unit): Unit = {
+    try {
+      DirectoryUtils.clean(esLocalPath)
+      File(s"$esLocalPath").mkdirs
+      unTar(Path.of(s"$localPath/elasticsearch.tar"), Path.of(s"$esLocalPath"))
+      File(s"$esLocalPath/plugins/readonlyrest/").mkdirs
+      Files.copy(
```

Comment:
I'm not sure why we do it like that? Cannot we just use the whole `/usr/share/elasticsearch` folder we downloaded?

Reply:
I remove it. It was needed, because before this iteration of review ROR plugin was not installed. Now it is installed and those files are present